grails-app/domain/icaruswings/Customer.groovy

```diff
@@ -1,18 +1,19 @@
 package icaruswings
 
 import icaruswings.utils.PersonType
+import icaruswings.utils.BasePerson
 
-class Customer extends icaruswings.utils.BaseEntity {
+class Customer extends BasePerson {
     String name
     String email
     String cpfCnpj
     PersonType personType
 
     static constraints = {
-        name nullable: true, blank: false
-        email nullable: true, blank: false, email: true
-        cpfCnpj nullable: true, blank: false, size: 11..14
-        personType nullable: true, blank: false
+        name nullable: false, blank: false
+        email nullable: false, blank: false, email: true
+        cpfCnpj nullable: false, blank: false, size: 11..18, unique: true
```

Comment:
Por que 11..18? Vejo que será bem mais simples se for salvo apenas os números e ao mostrar ser aplicado a formatação correta para cada dado.
Além disso deixando com essa quantidade de caracteres pode gerar inconsistências, pois será possível salvar um CNPJ com formatação e o mesmo CNPJ sem formatação e o sistema permitirá isso.